python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: The preprocessor used to transform data batches prior
+            to prediction.
+
+    """
+
+    def __init__(self, model: torch.nn.Module, preprocessor: Preprocessor):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition: If checkpoints
+                contains the model state dict, and not the model itself,
+                then the state dict will be loaded to the model_definition.
+
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        column_names: Optional[List[str]] = None,
+        feature_columns: Optional[Union[List[str], List[List[str]]]] = None,
+        dtype: Optional[Union[torch.dtype, List[torch.dtype]]] = None,
+    ) -> DataBatchType:
+        """Run inference on data batch.
+
+        Args:
+            data: A batch of input data.
+            columns_names: If provided, the column names to set for the data
+                batch.
+            feature_columns: The names of the columns in the dataframe to
```

Comment:
Either call this one `feature_column_names` or the other one `columns`, for consistency.

Also it says "dataframe" here, but my understanding is that we might have e.g. numpy arrays, too. Slightly confusing to me.

Also, is it a conscious decision not to include our native `Dataset` abstraction in all of this?

Reply:
+1 about first 2 points.

> Also, is it a conscious decision not to include our native Dataset abstraction in all of this?

Let's chat about this in a non-blocking way. 

My understanding is that we should look at datasets as a compute + data layer, and if you want to do large scale compute you should pass the model to dataset.map_batches. As such, we let predictors take single-threaded work, and the Dataset abstraction will handle scaling.

Reply:
`column_names` has been removed.

Changed the wording of dataframes.